src/components/SelectedAnnotation.jsx

```diff
@@ -323,11 +320,22 @@ class SelectedAnnotation extends React.Component {
     this.props.onClose();  //Note that deleteSelectedAnnotation() also runs unselectAnnotation(), but this needs to be called anyway to inform the parent component.
   }
 
+  removeEventListeners() {
+    this.inputText.removeEventListener('mouseup', () => {
+      this.dialog.className = ENABLE_DRAG;
+    });
+    this.inputText.removeEventListener('mousedown', () => {
+      this.dialog.className = DISABLE_DRAG;
+    });
+  }
+
   handleKeyUp(e) {
     if (Utility.getKeyCode(e) === KEY_CODES.ESCAPE) {
+      this.removeEventListeners();
```

Comment:
IIRC, you said that we had to remove the event listeners _before_ `componentWillUnmount` runs, because there's a bizarre situation where HTML elements are destroyed (in this case, `this.input`) _before_ the unmount process begins (instead of after it finishes, as expected), correct?

Did we figure out why? I'm still flabbergasted by this React WTF-ery.

Reply:
If it's confirmed that the issue is React WTF-ery, (and therefore we can't rely on `componentWillUnmount`) then we'll need to consider that it may be insufficient to just have `this.removeAllEventListener()` at the ESC call.

If we're going down this path, it would make more sense to
1. call `this.removeAllEventListeners()` before every instance of `this.props.onClose()` instead. (This would target _all_ user actions that gets rid of the transcription box)
2. fold `document.removeEventListener()` into `this.removeAllEventListener()`, since we're essentially saying that `this.removeAllEventListeners` is _replacing_ `componentWillUnmount()`

Reply:
I think the elements are removed before the unmount process begins because the `onClose` function runs and removes the dialog box before the child (`SelectedAnnotation`) has the opportunity to run the remove listener cleanup.